src/ostorlab/cli/scanner/scanner.py

```diff
@@ -2,17 +2,39 @@
 import asyncio
 import logging
 import sys
+import socket
+from typing import Optional
 
 import click
 
 from ostorlab.cli import console as cli_console
 from ostorlab.cli.rootcli import rootcli
 from ostorlab.scanner import start
+from ostorlab.utils import scanner_state_reporter
+from ostorlab.utils import ip
 
 console = cli_console.Console()
 
 logger = logging.getLogger(__name__)
 
+WAIT_CAPTURE_INTERVAL = 300
+
+
+async def _start_periodic_persist_state(
+    scanner_id: int, scan_id: Optional[int], errors: Optional[str]
+):
+    while True:
+        state_report = scanner_state_reporter.ScannerStateReporter(
+            scanner_id=int(scanner_id),
+            scan_id=scan_id,
+            hostname=socket.gethostname(),
+            ip=ip.get_public_ip(),
+            errors=errors,
```

Comment:
Errors and scan_id should be read from some state class. It should not be in the constructor of the reporter.